### DISCUSSION TITLE
Don't change language when config with localePrefix='never' and localeDetection = false

### DISCUSSION THREAD

**Discussion by navata** [amannn/next-intl#915] [Q&A]
### Description

I have download and run your demo with url https://github.com/amannn/next-intl/tree/main/examples/example-app-router
And i use config localeDetection = false in middleware, after function change language not working

**File define const:**

```
import {Pathnames} from 'next-intl/navigation';

export const locales = ['en', 'de'] as const;

export const pathnames = {
  '/': '/',
  '/pathnames': {
    en: '/pathnames',
    de: '/pfadnamen'
  }
} satisfies Pathnames<typeof locales>;

// Use the default: `always`
export const localePrefix = 'never';

export type AppPathnames = keyof typeof pathnames;

export const localeDetection = false;
```

**File middleware:**

```
import {pathnames, locales, localePrefix, localeDetection} from './config';
export default createMiddleware({
  localePrefix,
  localeDetection,
});
```
<img width="1538" alt="Screenshot 2024-03-03 at 16 37 47" src="https://github.com/amannn/next-intl/assets/16706514/abee9d53-55cb-425e-82e7-5ba96663b717">

### Mandatory reproduction URL

https://github.com/navata/examples-example-app-router

### Reproduction description

Steps to reproduce:
1. git clone https://github.com/navata/examples-example-app-router.git
2. npm i
3. npm run dev
4. change language on navbar -> it don't change language

### Expected behaviour

I want to change language when i use localePrefix='never' and localeDetection = false

**Comment by amannn** [comment]
If you use `localePrefix: 'never'` and `localeDetection: false`, the only remaining place where the locale can be resolved from is the domain, in case you use domain-based routing. If you use a single domain (or you have multiple locales per domain), you shouldn't disable `localeDetection`.

Can you share some background why you'd want to use this configuration?

**Comment by navata** [comment]
Thank you for your response. 
However, I would like the user to be able to control the language on my e-commerce website. All traffic will be directed to the same domain, which is advantageous for SEO. 
You can review prominent e-commerce websites such as https://shopee.vn/ or https://www.lazada.vn/.

**Comment by amannn** [comment]
shopee.vn unfortunately redirects me and I'm not able to use their language switcher. lazada.vn uses different domains for different locales, that's equivalent to domain-based routing in `next-intl`.

I'll move this to a discussion for the time being as this seems to be a usage question.
